src/odemis/driver/test/actuator_test.py

```diff
@@ -1117,30 +1122,50 @@ def test_offset_moveAbs(self):
         f = self.dev_cycle.moveAbs({axis_name: 0.0})
         f.result()
 
-        new_pos = 1.5707963267948966  # pi/2
-        for i in range(1, 6):
-            # overrun 2pi after 4 moves in clockwise
-            f = self.dev_cycle.moveRel({axis_name: new_pos})
+        # Check it runs referencing also after many small moves, by checking the time
+        durations = []
+        for i in range(self.ref_freq):
+            start_t = time.time()
+            f = self.dev_cycle.moveRelSync({axis_name: 0.1})
+            durations.append(time.time() - start_t)
+
+        # It should have 1 move which takes quite a lot longer (+1s)
+        dur_outlier = max(durations)
+        durations.remove(dur_outlier)
+        dur_std = sum(durations) / len(durations)
+        self.assertGreater(dur_outlier, dur_std + 1)
+
+        f = self.dev_cycle.moveAbs({axis_name: 0.0})
+        f.result()
+
+        # overrun 2pi after 4 moves in clockwise
+        shift = math.pi / 2
+        for i in range(5):
+            f = self.dev_cycle.moveRel({axis_name: shift})
             f.result()
-            if i == 5:
-                # if referencing was correct position should not differ more than by
-                # half of the ustepsize from the wanted position
-                self.assertLess(abs(self.dev_cycle.position.value[axis_name] - new_pos),
-                                self.dependency1._ustepsize[0] / 2.)
+        exp_pos = (shift * 5) % (2 * math.pi)  # = pi/2
+
+        self.assertAlmostEqual(self.dev_cycle.position.value[axis_name], exp_pos,
```

Comment:
You check here the position once reported by the wrapper class (RotationActuator) and once by the underlying HW (tmcm) right? I suggest to add a comment what these 2 tests do as it is not immediately obvious. 

`# Check positions reported by rotational axis is as expected after referencing and within the accuracy of the HW`
`# Check positions reported by linear axis (HW) is as expected after referencing and within the accuracy of the HW`

Q: Maybe I remember this incorrectly from our discussion, but I thought you mentioned that the controller does not know about "mistakes" in the physical position. That's why we regularly reference. If this is true, should we not remove the `_ustepsize` as accuracy measure? What do I miss?

Reply:
The ustepsize accuracy has nothing to do with the checked behaviour of the wrapper. However, the TMCM (simulator) still will not exactly move to where it was requested, it'll go there within a ustepsize, so it's important to use the ustepsize.